package HelloKittyMsgData;
import "common.proto";

enum BuildStatueType 
{
	 Build_Status_Normal = 1;
	 Build_Status_Wait_Click = 2;
	 Build_Status_Stop = 4;
}

message Point
{
	 required int32 x = 1;
	 required int32 y = 2;
}

message WareHouseBuildBase
{
	 required uint32 cellid = 1;
	 required uint32 type = 2;
	 required uint32 level = 3;
	 required uint32 status = 4;
	 required uint32 producetime = 5;
	 required uint32 durtime = 6;
	 required uint32 num = 7;
}

message BuildBase
{
	 required uint64 tempid = 1;
	 required uint32 type = 2;
	 required uint32 level = 3;
	 required Point point = 4;
	 required uint32 status = 5;
	 required uint32 durtime = 6;
	 required uint32 producetime = 7;
	 required bool rotationmark = 8 [default = false];
	 required bool active = 9 [default = false];
}

message ReqAllBuild
{
}

message AckFlushAllBuild
{
	 repeated BuildBase buildinfo = 1;
}

message ReqOneBuild
{
	 required uint64 tempid = 1;
}

message AckFlushOneBuild
{
	 required uint64 updatecharid = 1;
	 required BuildBase buildinfo = 2;
}

message ReqBuildUpGrade
{
	 required uint64 tempid = 1;
}

message ReqBuildMovePlace
{
	 required uint64 tempid = 1;
	 required Point point = 2;
	 required bool moveflg = 3 [default = true];
}

message ReqBuildBuilding
{
	 required uint32 type = 1;
	 required Point point = 2;
	 required bool moveflg = 3 [default = true];
}

message ReqBuildRoad
{
	 required uint32 type = 1;
	 repeated Point point = 2;
}

message ReqClearRoad
{
	 repeated uint64 tempid = 1;
}

enum BuildSuccessCodeType 
{
	 Build_Up_Grade = 0;
	 Build_Move = 1;
	 Build_Building = 2;
	 Build_Rotation = 3;
}

message AckBuildOpReturnSuccess
{
	 required uint64 updatecharid = 1;
	 required uint64 tempid = 2;
	 required BuildSuccessCodeType code = 3;
	 required BuildBase buildinfo = 4;
	 required Point oldpoint = 5;
}

enum BuildFailCodeType 
{
	 Build_Occcupy = 0;
	 Build_Premise_Limit = 1;
	 Build_Error = 2;
	 Build_Full_Level = 3;
	 Build_Type_Enough = 4;
}

message AckBuildOpReturnFail
{
	 required uint64 tempid = 1;
	 required ErrorCodeType commoncode = 2;
	 required BuildFailCodeType code = 3;
	 required BuildBase buildinfo = 4;
}

message ReqPickUpBuid
{
	 required uint64 tempid = 1;
}

message AckRemoveBuid
{
	 required uint64 updatecharid = 1;
	 required uint64 tempid = 2;
}

message ReqPickOutBuid
{
	 required uint32 cellid = 1;
	 required Point point = 2;
	 required bool moveflg = 3 [default = true];
}

message ReqClickRewardBuid
{
	 required uint64 tempid = 1;
}

message Worker
{
	 required uint32 id = 1;
	 required uint32 starttime = 2;
	 required uint32 cd = 3;
	 required uint32 lastsec = 4;
	 required uint32 build = 5;
}

message ReqWorker
{
}

message AckUpdateWorker
{
	 required Worker worker = 1;
}

message AckWorker
{
	 repeated Worker worker = 1;
}

message ReqResetWorker
{
	 required uint32 workerid = 1;
}

message ReqRecycleItem
{
	 required uint64 tempid = 1;
	 required uint32 itemid = 2;
	 required uint32 itemnum = 3;
}

message ProduceBase
{
	 required uint32 itemid = 1;
	 required uint32 value = 2;
}

message ReqBuildProduce
{
	 repeated uint64 tempid = 1;
}

message BuildProduce
{
	 required uint64 tempid = 1;
	 repeated ProduceBase produce = 2;
}

message AckBuildProduce
{
	 repeated BuildProduce produce = 1;
}

message AckUpdateRubbish
{
	 required uint64 updatecharid = 1;
	 required RubbishData rubbish = 2;
	 required uint32 optype = 3;
}

enum BufferSrcType 
{
	 BST_Item = 0;
	 BST_Build = 1;
	 BST_Divine = 2;
}

message BufferData
{
	 required BufferTypeID bufferid = 1;
	 required uint64 srcid = 2;
	 required BufferSrcType srctype = 3;
	 required uint32 begintime = 4;
	 required uint32 lasttime = 5;
	 required uint32 value = 6;
}

message BufferInfo
{
	 required uint64 tempid = 1;
	 repeated BufferData bufferdata = 2;
}

message AckUpdateBuffer
{
	 required BufferInfo bufferinfo = 1;
}
